### PR TITLE
Throw error on "return await" outside of try/catch.

### PR DESCRIPTION
**Pull request by mandyvenables**

### CODE REVIEW

**Comment by aljones15** [comment]
This seems like an improvement so thank you:

https://eslint.org/docs/rules/no-return-await

> Using return await inside an async function keeps the current function in the call stack until the Promise that is being awaited has resolved, at the cost of an extra microtask before resolving the outer Promise. return await can also be used in a try/catch statement to catch errors from another function that returns a Promise.
> 
> You can avoid the extra microtask by not awaiting the return value, with the trade off of the function no longer being a part of the stack trace if an error is thrown asynchronously from the Promise being returned. This can make debugging more difficult.

**Review by aljones15 — APPROVED** [review]
this looks good pending @mattcollier or @davidlehn  approving I think we can add this rule. Thanks for contributing. Also there are some issues with other rule suggestions we looking for consensus on if you want to take a look.

File: index.js [hunk 1]
```diff
@@ -41,6 +41,7 @@ module.exports = {
     'no-multi-spaces': 'error',
     'no-multiple-empty-lines': ['error', {max: 1}],
     'no-spaced-func': 'error',
+    'no-return-await': 'error',
```
Comment (davidlehn) [inline index.js:44]: use alpha order
```suggestion
    'no-return-await': 'error',
    'no-spaced-func': 'error',
```

**Comment by davidlehn** [comment]
I think this is fine since "return await" usually isn't desired.  @aljones15 I'm not sure what your stack trace example is doing.  Note that modern v8 (and maybe other runtimes) are pretty good at optimizing variations of these patterns and providing useful stack traces.  You don't usually need to jump through hoops to do so.

**Review by davidlehn — APPROVED** [review]

**Pull request merged**